### ISSUE TITLE
[BUG] Failed to upgrade harvester when node name too long cause error "Invalid value: ... must be no more than 63 characters"

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#2114]
**Describe the bug**

When upgrade harvester from v1.0.0 to v1.0.1-rc2, if the node name too long like `harvester-101upgrade-node1`
The upgrade will fail with error message
  ```
  Job.batch "hvst-upgrade-grwbw-single-node-upgrade-harvester-101upgrade-node1" is invalid: spec.template.labels: 
  Invalid value: "hvst-upgrade-grwbw-single-node-upgrade-harvester-101upgrade-node1": must be no more than 63 characters
  ```
  ![image](https://user-images.githubusercontent.com/29251855/161706107-a54d33a0-9198-4d41-87db-8af48bfe6dee.png)

**To Reproduce**
1. Prepare v1.0.0 one node harvester
2. Set a longer node name `harvester-101upgrade-node1`
3. Create a ubuntu image
4. Create a virtual machine
5. Shutdown VM
6. Follow the steps to upgrade to v1.0.1-rc2
https://github.com/harvester/docs/blob/42598035cfe79856f7e159d4c47c4d7d29178900/docs/upgrade/automatic.md

**Expected behavior**

Should allow longer node name to prevent the no longer than 63 characters error during the upgrade to v1.0.1

**Support bundle**

[supportbundle_37fc8ce4-a657-4380-b86d-b517ee995c0e_2022-04-05T07-52-07Z.zip](https://github.com/harvester/harvester/files/8419019/supportbundle_37fc8ce4-a657-4380-b86d-b517ee995c0e_2022-04-05T07-52-07Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.0 GA to v1.0.1-rc2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 1 node harvester

**Additional context**

**Comment by bk201** [comment]
https://github.com/harvester/harvester/blob/03ee62133765a03394e1a35ccd4859ad1afd1302/pkg/controller/master/upgrade/common.go#L200
Here the job name is automatically set to label `job-name` by Kubernetes. We need a way to handle the name length > 63 characters. 

A possible way to fix is follow how SUC define a plan's job names: https://github.com/rancher/system-upgrade-controller/blob/79da9f058209b9906bfd6163d58672e9ae67ea5d/pkg/upgrade/job/job.go#L103

**Comment by TachunLin** [comment]
Verified **fixed** after upgrade from `v1.0.1` to `master-0eab7caa-head` (04/24, v1.0.1). Close this issue.

### Result
We can upgrade harvester when node name `harvester-101upgrade-node1` too long (25 characters)
  ![image](https://user-images.githubusercontent.com/29251855/165072808-8bdab562-6e17-422d-b981-ed8e9ff25055.png)
  ![image](https://user-images.githubusercontent.com/29251855/165073479-29923b79-a931-493f-8fad-44af1220cc2a.png)

* Upgrade job name
```
harvester-101upgrade-node1:~ # kubectl get jobs -A | grep hvst
cattle-system      apply-hvst-upgrade-lvxs4-prepare-on-harvester-101upgrade-fa83a5   1/1           4m52s      15m
harvester-system   hvst-upgrade-lvxs4-apply-manifests                                1/1           2m12s      10m
harvester-system   hvst-upgrade-lvxs4-single-node-upgrade-harvester-101upgra-8c248   1/1           2m9s       8m46s
```


### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: `v1.0.1` to `master-0eab7caa-head` (04/24)

### Verify Steps
1. Prepare v1.0.0 one node harvester
2. Set a longer node name `harvester-101upgrade-node1`
3. Create a ubuntu image
4. Create a virtual machine
5. Shutdown VM
6. Follow the steps to upgrade to v1.0.1-rc2
https://github.com/harvester/docs/blob/42598035cfe79856f7e159d4c47c4d7d29178900/docs/upgrade/automatic.md
